<?xml version='1.0'?>
<sdf version='1.4'>
	<!-- http://gazebosim.org/tutorials?tut=build_robot&cat=build_robot -->
	<model name="simple_diff">
		<static>false</static>
		 
		<link name='chassis'>
			<pose>0 0 .1 0 0 0</pose>

			<inertial>
				<mass>5</mass>
				<pose>-.1 0 -.1 0 0 0</pose>
			</inertial>
			
			<collision name='collision'>
				<geometry>
				<box>
					<size>.4 .2 .1</size>
				</box>
				</geometry>
			</collision>

			<visual name='visual'>
				<geometry>
				<box>
					<size>.4 .2 .1</size>
				</box>
				</geometry>
				<material>
					<ambient>0.5 0.5 0.5 1</ambient>
					<diffuse>0.0 0.85 0.25 1</diffuse>
					<specular>0 0 0 0</specular>
					<emissive>0 0 0 1</emissive>
				</material>
			</visual>

			<collision name='caster_collision'>
				<pose>-0.15 0 -0.05 0 0 0</pose>
				<geometry>
				<sphere>
					<radius>.05</radius>
				</sphere>
				</geometry>

				<surface>
					<friction>
						<ode>
							<mu>0</mu>
							<mu2>0</mu2>
							<slip1>1.0</slip1>
							<slip2>1.0</slip2>
						</ode>
					</friction>
				</surface>
			</collision>

			<visual name='caster_visual'>
				<pose>-0.15 0 -0.05 0 0 0</pose>
				<geometry>
					<sphere>
					<radius>.05</radius>
					</sphere>
				</geometry>
			</visual>
		</link>
		
		<link name="left_wheel">
			<pose>0.1 0.13 0.1 0 1.5707 1.5707</pose>
			<collision name="collision">
				<geometry>
					<cylinder>
						<radius>.1</radius>
						<length>.05</length>
					</cylinder>
				</geometry>
			</collision>
			<visual name="visual">
				<geometry>
					<cylinder>
						<radius>.1</radius>
						<length>.05</length>
					</cylinder>
				</geometry>
				<material>
					<ambient>0.35 0.35 0.35 1</ambient>
					<diffuse>0.45 0.45 0.45 1</diffuse>
					<specular>0 0 0 0</specular>
					<emissive>0 0 0 1</emissive>
				</material>
			</visual>
		</link>
		
		<link name="right_wheel">
			<pose>0.1 -0.13 0.1 0 1.5707 1.5707</pose>
			<collision name="collision">
				<geometry>
					<cylinder>
						<radius>.1</radius>
						<length>.05</length>
					</cylinder>
				</geometry>
			</collision>
			<visual name="visual">
				<geometry>
					<cylinder>
						<radius>.1</radius>
						<length>.05</length>
					</cylinder>
				</geometry>
				<material>
					<ambient>0.35 0.35 0.35 1</ambient>
					<diffuse>0.45 0.45 0.45 1</diffuse>
					<specular>0 0 0 0</specular>
					<emissive>0 0 0 1</emissive>
				</material>
			</visual>
		</link>
		
		<joint type="revolute" name="left_wheel_hinge">
			<pose>0 0 -0.03 0 0 0</pose>
			<child>left_wheel</child>
			<parent>chassis</parent>
			<axis>
				<xyz>0 1 0</xyz>
			</axis>
		</joint>

		<joint type="revolute" name="right_wheel_hinge">
			<pose>0 0 0.03 0 0 0</pose>
			<child>right_wheel</child>
			<parent>chassis</parent>
			<axis>
				<xyz>0 1 0</xyz>
			</axis>
		</joint>
		
		<link name="camera_link">
			<gravity>false</gravity>
			<pose>.175 0 .2 0 0.1 0</pose>
			<inertial>
				<mass>0.1</mass>
			</inertial>
			<visual name="visual">
				<geometry>
					<box>
						<size>0.05 0.05 0.05</size>
					</box>
				</geometry>
				<material>
					<ambient>0.75 0.75 0.75 1</ambient>
					<diffuse>0.8 0.85 0.8 1</diffuse>
					<specular>0 0 0 0</specular>
					<emissive>0 0 0 1</emissive>
				</material>
			</visual>
			<sensor name="camera" type="camera">
				<camera>
					<!--<horizontal_fov>1.047</horizontal_fov>
					<!--<horizontal_fov>2.79253</horizontal_fov>-->
					<horizontal_fov>2.0</horizontal_fov>
					<image>
						<width>320</width>
						<height>240</height>
						<format>R8G8B8</format>
					</image>
					
					<!--<clip>
						<near>0.01</near>
						<far>10</far>
					</clip>-->
				
					<!--<noise>
						<type>gaussian</type>
						<mean>0.0</mean>
						<stddev>0.007</stddev>
					</noise>-->
				</camera>
				<always_on>1</always_on>
				<update_rate>10</update_rate>
				<visualize>true</visualize>
			</sensor>
		</link>

		<joint name="camera_joint" type="fixed">
			<parent>chassis</parent>
			<child>camera_link</child>
		</joint>
	</model>
</sdf>